### PR TITLE
Adds COP for heating and hot water

### PR DESCRIPTION
**Pull request by kr0ner**

### CODE REVIEW

File: yaml/common.yaml [hunk 1]
```diff
@@ -162,6 +162,30 @@ sensor:
...
+#                                       #
+#   Template Sensors                    #
+#                                       #
+#########################################
+  - platform: template
+    name: "COP WW TAG"
+    update_interval: $interval_very_slow
+    state_class: "measurement"
+    lambda: |-
+      if (id(WAERMEERTRAG_WW_TAG_SUMME_KWH).state && id(EL_AUFNAHMELEISTUNG_WW_SUMME_KWH).state)
```
Comment (mkaiser) [inline yaml/common.yaml:175]: add a check for non-zero demoninators: 

if (id(WAERMEERTRAG_WW_TAG_SUMME_KWH).state && id(EL_AUFNAHMELEISTUNG_WW_SUMME_KWH).state && id(EL_AUFNAHMELEISTUNG_WW_SUMME_KWH).state != 0)

**Comment by mkaiser** [comment]
I find this a bit confusing with the SUMME_KWH vs. SUMME_MWH endings - can't you just remove the postfix (marked in bold) here for the "logical" sensors in the yaml?

"WAERMEERTRAG_2WE_WW_**SUMME_MWH**:   !include { file: wp_generic_combined.yaml, vars: { sensor_name: "WAERMEERTRAG_2WE_WW_**SUMME_MWH**"      , scaled_property: "WAERMEERTRAG_2WE_WW_SUM_KWH"  , property: "WAERMEERTRAG_2WE_WW_**SUM_MWH**"  , unit: "MWh", accuracy_decimals: "3", icon: "mdi:fire"  }}
  WAERMEERTRAG_2WE_HEIZ_**SUMME_MWH**: !include { file: wp_generic_combined.yaml, vars: { sensor_name: "WAERMEERTRAG_2WE_HEIZ_**SUMME_MWH**"    , scaled_property: "WAERMEERTRAG_2WE_HEIZ_SUM_KWH", property: "WAERMEERTRAG_2WE_HEIZ_**SUM_MWH**", unit: "MWh", accuracy_decimals: "3", icon: "mdi:fire"  }}
  WAERMEERTRAG_WW_**SUMME_MWH:**       !include { file: wp_generic_combined.yaml, vars: { sensor_name: "WAERMEERTRAG_WW_**SUMME_MWH**"          , scaled_property: "WAERMEERTRAG_WW_SUM_KWH"      , property: "WAERMEERTRAG_WW_**SUM_MWH**"      , unit: "MWh", accuracy_decimals: "3", icon: "mdi:fire"  }}
  WAERMEERTRAG_HEIZ_**SUMME_MWH**:     !include { file: wp_generic_combined.yaml, vars: { sensor_name: "WAERMEERTRAG_HEIZ_**SUMME_MWH**"        , scaled_property: "WAERMEERTRAG_HEIZ_SUM_KWH"    , property: "WAERMEERTRAG_HEIZ_**SUM_MWH**"    , unit: "MWh", accuracy_decimals: "3", icon: "mdi:fire"  }}

... and the same for the electrical energy EL_AUFNAHMELEISTUNG  (yeah, I promised the PR to rename it to EL_AUFNAHMELEISTUNG, but I like to get my stuff working before)

Also maybe change SUMME to TOTAL?


I  am really confused :)

**Comment by mkaiser** [comment]
Another thing coming to my mind: 

For the template update interval "$interval_very_slow" is used.

wp_daily_energy_combined.yaml has no interval set and uses the default update interval of 60s. 

I suggest to skip the interval for the templates and use the default of 60 s

**Comment by kr0ner** [comment]
> I find this a bit confusing with the SUMME_KWH vs. SUMME_MWH endings - can't you just remove the postfix (marked in bold) here for the "logical" sensors in the yaml? (see https://github.com/mkaiser/OneESP32ToRuleThemAll/blob/5d22afd6af31382cffb12075dc4e55950c0c8c01/yaml/ttf07.yaml#L226)
> 
> "WAERMEERTRAG_2WE_WW_**SUMME_MWH**: !include { file: wp_generic_combined.yaml, vars: { sensor_name: "WAERMEERTRAG_2WE_WW_**SUMME_MWH**" , scaled_property: "WAERMEERTRAG_2WE_WW_SUM_KWH" , property: "WAERMEERTRAG_2WE_WW_**SUM_MWH**" , unit: "MWh", accuracy_decimals: "3", icon: "mdi:fire" }} WAERMEERTRAG_2WE_HEIZ_**SUMME_MWH**: !include { file: wp_generic_combined.yaml, vars: { sensor_name: "WAERMEERTRAG_2WE_HEIZ_**SUMME_MWH**" , scaled_property: "WAERMEERTRAG_2WE_HEIZ_SUM_KWH", property: "WAERMEERTRAG_2WE_HEIZ_**SUM_MWH**", unit: "MWh", accuracy_decimals: "3", icon: "mdi:fire" }} WAERMEERTRAG_WW_**SUMME_MWH:** !include { file: wp_generic_combined.yaml, vars: { sensor_name: "WAERMEERTRAG_WW_**SUMME_MWH**" , scaled_property: "WAERMEERTRAG_WW_SUM_KWH" , property: "WAERMEERTRAG_WW_**SUM_MWH**" , unit: "MWh", accuracy_decimals: "3", icon: "mdi:fire" }} WAERMEERTRAG_HEIZ_**SUMME_MWH**: !include { file: wp_generic_combined.yaml, vars: { sensor_name: "WAERMEERTRAG_HEIZ_**SUMME_MWH**" , scaled_property: "WAERMEERTRAG_HEIZ_SUM_KWH" , property: "WAERMEERTRAG_HEIZ_**SUM_MWH**" , unit: "MWh", accuracy_decimals: "3", icon: "mdi:fire" }}
> 
> ... and the same for the electrical energy EL_AUFNAHMELEISTUNG (yeah, I promised the PR to rename it to EL_AUFNAHMELEISTUNG, but I like to get my stuff working before)
> 
> Also maybe change SUMME to TOTAL?
> 
> I am really confused :)

Yes, makes sense to remove them ... they anyways have a unit ... makes no sense to put it in the name again

**Comment by kr0ner** [comment]
> Another thing coming to my mind:
> 
> For the template update interval "$interval_very_slow" is used.
> 
> wp_daily_energy_combined.yaml has no interval set and uses the default update interval of 60s.
> 
> I suggest to skip the interval for the templates and use the default of 60 s

Actually it should be set to "never" for the sake of making it obvious. #56
Since there is no lambda, nothing will happen anyways.
The value is updated every 5 minutes via schedule.
>  Every hour, every 5 minutes except midnight.

**Pull request merged**